build-system/pr-check/build-targets.js

```diff
@@ -252,6 +281,12 @@ function determineBuildTargets(fileName = 'build-targets.js') {
         .join(', ')
     )
   );
+  if (isTravisBuild() && buildTargets.has('PACKAGE_UPGRADE')) {
+    console.log(
+      `${fileLogPrefix} Since this PR contains package upgrade(s), running all checks...`
+    );
```

Comment:
I would leave the logging to the Travis job scripts themselves, and let this file be a util to determine build targets.

Reply:
Thought about that, but there are ten call-sites for this, and each of them will have to individually extract the full list of targets and replicate the same logic.

Reply:
Does `PACKAGE_UPGRADE` need to be a build target? What if we checked for package upgrades, then added all the build targets, and ran through the build without any special logging?

Reply:
Good question. The reason I added a new target was because in future, we may want to run other checks specific to just the main `package.json`.

For now, I'm happy to do what you suggest, or leave things as they are. Let me know your preference.

Reply:
Ideally we could keep `build-target.js` a util file and let the Travis job scripts control the logging.

Getting rid of `PACKAGE_UPGRADE` could be a solution , but if you want to keep it, we can move the logging to the scripts, although it would be repetitive, or not log at all.

Reply:
I agree that logging about tests is weird in this file. I've removed the line because I don't think it's all that necessary. The various test / check logs should be self explanatory.